src/components/CalendarContainer.ts

```diff
@@ -1,14 +1,215 @@
+
+        window.mx.data.get({
+            xpath: XPath,
+            callback: mxObjects => this.setEventsFromMxObjects(mxObjects)
+        });
+    }
+
+    private setEventsFromMxObjects = (mxObjects: mendix.lib.MxObject[]) => {
+        const { titleAttribute, startAttribute, endAttribute } = this.props;
+        const events = mxObjects.map(mxObject => {
+            const title = mxObject.get(titleAttribute) as string;
+            const start = new Date(mxObject.get(startAttribute) as number);
+            const end = new Date(mxObject.get(endAttribute) as number);
+            const guid = mxObject.getGuid();
+
+            return {
+                title,
+                start,
+                end,
+                guid
+            };
+        });
+        this.setState({ events });
+    }
+
+    private fetchEventsByMicroflow(mxObject?: mendix.lib.MxObject) {
+        const { dataSourceMicroflow } = this.props;
+        if (mxObject && dataSourceMicroflow) {
+            window.mx.ui.action(dataSourceMicroflow, {
+                callback: (mxObjects: mendix.lib.MxObject[]) => this.setEventsFromMxObjects(mxObjects),
+                error: error => window.mx.ui.error(`Error while executing microflow: ${dataSourceMicroflow}: ${error.message}`),
+                params: {
+                    applyto: "selection",
+                    guids: mxObject ? [ mxObject.getGuid() ] : []
+                }
+            });
+        }
+    }
+
+    private excecuteEventAction = () => {
+        const { mxObject, onClickEvent, eventMicroflow, mxform, eventNanoflow, openPageAs, eventPage } = this.props;
+        if (!mxObject || !mxObject.getGuid()) {
+            return;
+        }
+        const context = new mendix.lib.MxContext();
+        context.setContext(mxObject.getEntity(), mxObject.getGuid());
+        if (onClickEvent === "callMicroflow" && eventMicroflow && mxObject.getGuid()) {
+            window.mx.ui.action(eventMicroflow, {
+                error: error => window.mx.ui.error(`Error while executing microflow: ${eventMicroflow}: ${error.message}`),
+                origin: mxform,
+                context
+            });
+        } else if (onClickEvent === "callNanoflow" && eventNanoflow.nanoflow) {
+            window.mx.data.callNanoflow({
+                context,
+                error: error => window.mx.ui.error(
+                    `An error occurred while executing the nanoflow: ${error.message}`
+                ),
+                nanoflow: eventNanoflow,
+                origin: mxform
+            });
+        } else if (onClickEvent === "showPage" && eventPage && mxObject.getGuid()) {
+            window.mx.ui.openForm(eventPage, {
+                context,
+                error: error => window.mx.ui.error(
+                    `Error while opening page ${eventPage}: ${error.message}`
+                ),
+                location: openPageAs
+            });
```

Comment:
Arrange properties in the same order as its done with in the mendix api

Reply:
Done